src/App.js

```diff
@@ -1,17 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import TimeStamp from './components/TimeStamp';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatMessagesData, setChatMessagesData] = useState(chatMessages);
+  const [likesNumber, setLikesNumber] = useState(0);
+
+  const updateLiked = (updatedChatData) => {
+    const chats = chatMessagesData.map((chat) => {
+      if (chat.id === updatedChatData.id) {
+        return updatedChatData;
+      } else {
+        return chat;
+      }
+    });
+    const countLikes = chats.filter((chatComp) => chatComp.liked).length;
+    setLikesNumber(countLikes);
+    setChatMessagesData(chats);
+  };
+
+  const names = Array.from(
+    new Set(
+      chatMessagesData.map((chat) => {
+        return chat.sender;
+      })
+    )
+  );
+
   return (
     <div id="App">
-      <header>
-        <h1>Application title</h1>
-      </header>
-      <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
-      </main>
+      <React.StrictMode>
```

Comment:
We have this established in `index.js`, so I don't think we need to have it here, too. But I could be wrong!